crypto/keypair.go

```diff
@@ -19,11 +20,12 @@ type PrivateKey interface {
 	Decode([]byte) error
 }
 
-func DecodePrivateKey(in []byte) (PrivateKey, error) {
-	priv, err := NewEd25519PrivateKey(in)
-	if err != nil {
-		return nil, err
+func DecodePrivateKey(in []byte, keytype string) (priv PrivateKey, err error) {
```

Comment:
Why is this necessary if keys have a `Decode` method? Instantiating the type doesn't seem any more cumbersome than specifying the type as a string.

Reply:
it's used when decoding a private key from an encrypted keystore, since in that case the type is saved as a string inside the file. when we decode the key, it's just raw bytes, we don't have a decode method for it.

Reply:
see DecryptPrivateKey in keystore/keystore.go

Reply:
I have made the updates that add KeyType